src/controllers/QuestionnaireController.ts

```diff
@@ -51,4 +53,120 @@ export class QuestionnaireController {
             }
         );
     }
+
+    /**
+     * Add a questionnaire.
+     *
+     * @param {Request} req
+     * @param {Response} res
+     * @memberof QuestionnaireController
+     */
+    @Post('')
+    @Middleware(
+        jwt({
+            secret: AuthConfig.jwtSecret,
+            algorithms: ['HS256'],
+            requestProperty: 'payload',
+            isRevoked: AuthorizationController.checkApiUserLogin
+        })
+    )
+    public async addQuestionnaire(req: Request, res: Response) {
+        const url = req.body.url;
+        const version = req.body.version;
+        const name = req.body.name;
+        const questionnaire = req.body.questionnaire;
+
+        this.questionnaireModel.addQuestionnaire(url, version, name, questionnaire).then(
+            () => {
+                res.status(200).send();
+            },
+            (err) => {
+                if (err.code === '409') {
+                    res.status(409).send('A questionnaire with this name already exists.');
+                } else {
+                    res.status(500).send();
+                }
+            }
+        );
+    }
+
+    /**
+     * Update a questionnaire.
+     *
+     * @param {Request} req
+     * @param {Response} res
+     * @memberof QuestionnaireController
+     */
+    @Put('')
+    @Middleware(
+        jwt({
+            secret: AuthConfig.jwtSecret,
+            algorithms: ['HS256'],
+            requestProperty: 'payload',
+            isRevoked: AuthorizationController.checkApiUserLogin
+        })
+    )
+    public async updateQuestionnaire(req: Request, res: Response) {
+        const url = req.body.url;
```

Comment:
Same issue

Reply:
No, it is the other way round: [req.query](https://expressjs.com/de/api.html#req.query) and[ req.body](https://expressjs.com/de/api.html#req.body) ?